### PR TITLE
Continue on Develop before creating rc-* branch

### PR DESCRIPTION
**Pull request by CJ42**
Merge latest change (see below) onto develop

- [x] `KeyManager` with MultiChannel nonce + tests
- [x] Update README with installations + usage example
- [x] Updates related to `ERC725Y` breaking changes (setData array) 
    - created LSP2 Utils library (for `getDataSingle(…)`)
    - applied to contracts implementation for LSP3 and LSP4.
    - `KeyManager` loop through keys to check permissions
- [x] Package-lock.json version 2 (NPM 7)
- [x] Clean Solidity code + comments

Next to do:

- [ ]  split **KeyManager** ✅ , LSP3Account and UniversalReceiver contract between Core and normal / Init
- [x] change of parameters in signatures (_data) as 3rd param + refactor unit tests with new order
- [x] changed order of signature (flipped abi <> nonce) in `executeRelayCall`

### CODE REVIEW

File: contracts/Factories/Create2Factory.sol [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
...ffffffffffffffffffffff191681526001018473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1660601b81526014018381526020018281526020019450505050506040516020818303038152906040528051906020012090508060001c9150509392505050565b60008060003490506000845114156103c6576040517f08c379a00000000000000000000000000000000000000000000000000000000081526004018080602001828103825260208152602001807f437265617465323a2062797465636f6465206c656e677468206973207a65726f81525060200191505060405180910390fd5b8484516020860183f59150600073ffffffffffffffffffffffffffffffffffffffff168273ffffffffffffffffffffffffffffffffffffffff161415610474576040517f08c379a00000000000000000000000000000000000000000000000000000000081526004018080602001828103825260198152602001807f437265617465323a204661696c6564206f6e206465706c6f790000000000000081525060200191505060405180910390fd5b7fc16bb3dbd36917c7aa3e76b988c2cd35e74bb230a02fef61e7376d8b4bfaea778286604051808373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020018281526020019250505060405180910390a181925050509291505056fea26469706673582212208e45d684157bfc10076a3d284f8c3899cd5ecf31abade0a086223e0886a8a68164736f6c63430006010033
```
Comment (frozeman) [inline contracts/Factories/Create2Factory.sol:7]: This needs to be added back!

File: contracts/Factories/Create2Factory.sol [hunk 2]
```diff
@@ -1,21 +1,10 @@
...
...526014018381526020018281526020019450505050506040516020818303038152906040528051906020012090508060001c9150509392505050565b60008060003490506000845114156103c6576040517f08c379a00000000000000000000000000000000000000000000000000000000081526004018080602001828103825260208152602001807f437265617465323a2062797465636f6465206c656e677468206973207a65726f81525060200191505060405180910390fd5b8484516020860183f59150600073ffffffffffffffffffffffffffffffffffffffff168273ffffffffffffffffffffffffffffffffffffffff161415610474576040517f08c379a00000000000000000000000000000000000000000000000000000000081526004018080602001828103825260198152602001807f437265617465323a204661696c6564206f6e206465706c6f790000000000000081525060200191505060405180910390fd5b7fc16bb3dbd36917c7aa3e76b988c2cd35e74bb230a02fef61e7376d8b4bfaea778286604051808373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020018281526020019250505060405180910390a181925050509291505056fea26469706673582212208e45d684157bfc10076a3d284f8c3899cd5ecf31abade0a086223e0886a8a68164736f6c634300060100331ba01234123412341234123412341234123412341234123412341234123412341234a01234123412341234123412341234123412341234123412341234123412341234
```
Comment (frozeman) [inline contracts/Factories/Create2Factory.sol:14]: same

File: contracts/Factories/Create2Factory.sol [hunk 3]
```diff
@@ -1,21 +1,10 @@
 // SPDX-License-Identifier: MIT
-// based on https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Create2.sol
-
```
Comment (frozeman) [inline contracts/Factories/Create2Factory.sol:3]: Please add
`// DO NOT TOUCH`

File: contracts/KeyManager/KeyManager.sol [hunk 4]
```diff
@@ -136,18 +164,17 @@ contract KeyManager is ERC165, IERC1271 {
...
 
-        // recover the signer
         address from = keccak256(blob).toEthSignedMessageHash().recover(
             _signature
         );
-        // require(hasRole(EXECUTOR_ROLE, from), 'Only executors are allowed');
+
         require(
-            _nonceStore[from] == _nonce,
+            _verifyNonce(from, _nonce),
```
Comment (frozeman) [inline contracts/KeyManager/KeyManager.sol:172]: Whats the max allows error message length?

Comment (frozeman) [reply contracts/KeyManager/KeyManager.sol:172]: https://medium.com/@chebyk.in/how-big-is-solidity-custom-error-messages-overhead-1e915724b450

File: contracts/Factories/Create2Factory.sol (hunk 3 above)
Comment (CJ42) [reply contracts/Factories/Create2Factory.sol:3]: Done

File: contracts/KeyManager/KeyManager.sol (hunk 4 above)
Comment (CJ42) [reply contracts/KeyManager/KeyManager.sol]: @frozeman changed linter to 62 during our discussion today.

File: contracts/Factories/Create2Factory.sol (hunk 1 above)
Comment (CJ42) [reply contracts/Factories/Create2Factory.sol:7]: @frozeman added back in `rc-0.1.4` branch

File: contracts/Factories/Create2Factory.sol (hunk 2 above)
Comment (CJ42) [reply contracts/Factories/Create2Factory.sol:14]: @frozeman added back in `rc-0.1.4` branch

**Pull request merged**